--- origsrc/xfce4-indicator-plugin-2.3.3/panel-plugin/indicator-box.c	2014-04-29 16:12:51.000000000 -0500
+++ src/xfce4-indicator-plugin-2.3.3/panel-plugin/indicator-box.c	2016-12-02 13:55:21.702249600 -0600
@@ -185,7 +185,7 @@ xfce_indicator_box_sort_buttons (gconstp
 
   /* special case for Application indicators (unreliable ordering) */
   /* always compare by name and ignore location field */
-  if (a_io != NULL && g_strcmp0 (a_io, "libapplication.so") == 0)
+  if (a_io != NULL && g_strcmp0 (a_io, LIBSO("application")) == 0)
     result = g_strcmp0 (xfce_indicator_button_get_entry(a0)->name_hint,
                         xfce_indicator_button_get_entry(b0)->name_hint);
 
@@ -224,7 +224,7 @@ xfce_indicator_box_add (GtkContainer *co
   io_name = xfce_indicator_button_get_io_name (button);
   li = g_hash_table_lookup (box->children, io_name);
   // printf ("   +++ %s %s\n", io_name, xfce_indicator_button_get_entry (button)->name_hint);
-  if (g_strcmp0 (io_name, "libapplication.so") != 0 &&
+  if (g_strcmp0 (io_name, LIBSO("application")) != 0 &&
       xfce_indicator_button_get_pos (button) == 0)
     li = g_list_append (li, button);
   else
--- origsrc/xfce4-indicator-plugin-2.3.3/panel-plugin/indicator-config.h	2014-05-05 14:51:24.000000000 -0500
+++ src/xfce4-indicator-plugin-2.3.3/panel-plugin/indicator-config.h	2016-12-02 14:01:12.673703400 -0600
@@ -22,6 +22,12 @@
 #include <glib.h>
 #include <dbus/dbus-glib.h>
 
+#ifdef __CYGWIN__
+#define LIBSO(name) "cyg" name ".dll"
+#else
+#define LIBSO(name) "lib" name ".so"
+#endif
+
 G_BEGIN_DECLS
 
 typedef struct _IndicatorConfigClass IndicatorConfigClass;
--- origsrc/xfce4-indicator-plugin-2.3.3/panel-plugin/indicator-dialog.c	2014-06-03 17:45:12.000000000 -0500
+++ src/xfce4-indicator-plugin-2.3.3/panel-plugin/indicator-dialog.c	2016-12-02 13:55:21.757252900 -0600
@@ -54,26 +54,26 @@
 static const gchar *pretty_names[][3] =
 {
   /* raw name,                           pretty name,                                 icon-name(?) */
-  { "libapplication.so",                 N_("Application Indicators"),               "application-default-icon" },
-  { "libsoundmenu.so",                   N_("Sound Menu"),                           "preferences-desktop-sound" },
+  { LIBSO("application"),                 N_("Application Indicators"),               "application-default-icon" },
+  { LIBSO("soundmenu"),                   N_("Sound Menu"),                           "preferences-desktop-sound" },
   { "com.canonical.indicator.sound",     N_("Sound Menu"),                           "preferences-desktop-sound" },
-  { "libprintersmenu.so",                N_("Printers Menu"),                        "printer" },
-  { "libpower.so",                       N_("Power Management"),                     NULL },
+  { LIBSO("printersmenu"),                N_("Printers Menu"),                        "printer" },
+  { LIBSO("power"),                       N_("Power Management"),                     NULL },
   { "com.canonical.indicator.power",     N_("Power Management"),                     NULL },
-  { "libappmenu.so",                     N_("Application Menus (Global Menu)"),      "menu-editor" },
-  { "libsession.so",                     N_("Session Management"),                   NULL },
+  { LIBSO("appmenu"),                     N_("Application Menus (Global Menu)"),      "menu-editor" },
+  { LIBSO("session"),                     N_("Session Management"),                   NULL },
   { "com.canonical.indicator.session",   N_("Session Management"),                   NULL },
-  { "libmessaging.so",                   N_("Messaging Menu"),                       "indicator-messages-new" },
+  { LIBSO("messaging"),                   N_("Messaging Menu"),                       "indicator-messages-new" },
   { "com.canonical.indicator.messages",  N_("Messaging Menu"),                       "indicator-messages-new" },
-  { "libdatetime.so",                    N_("Date and Time"),                        "time-admin" },
+  { LIBSO("datetime"),                    N_("Date and Time"),                        "time-admin" },
   { "com.canonical.indicator.datetime",  N_("Date and Time"),                        "time-admin" },
-  { "libbluetooth.so",                   N_("Bluetooth"),                            "bluetooth-active" },
+  { LIBSO("bluetooth"),                   N_("Bluetooth"),                            "bluetooth-active" },
   { "com.canonical.indicator.bluetooth", N_("Bluetooth"),                            "bluetooth-active" },
-  { "libsyncindicator.so",               N_("Sync Menu"),                            "sync-client-updating" },
+  { LIBSO("syncindicator"),               N_("Sync Menu"),                            "sync-client-updating" },
   { "com.canonical.indicator.keyboard",  N_("Keyboard"),                             NULL },
   { "com.canonical.indicator.location",  N_("Location"),                             NULL },
   { "com.canonical.indicator.network",   N_("Network"),                              NULL },
-  { "libworkrave.so",                    N_("Workrave"),                             "workrave" },
+  { LIBSO("workrave"),                    N_("Workrave"),                             "workrave" },
 };
 
 
